syntaxes/src/expression.ts

```diff
@@ -594,28 +594,15 @@ export const Expression: GrammarDefinition = {
     },
 
     ternaryExpression: {
-      begin: /(?=\?)(?!\?\.)/,
-      end: /(?=$|"|[;,})\]])/,
-      patterns: [
-        {
-          include: '#ternaryOperator',
-        },
-        {
-          include: '#ngExpression',
-        },
-      ],
-    },
-
-    ternaryOperator: {
-      begin: /(\?)(?!\.)/,
+      begin: /(?!\?\.\s*[^[:digit:]])(\?)(?!\?)/,
```

Comment:
Since this file is a TS file, could you add a comment explaining the regex groups:
1. negative lookahead `(?!\?\.\s*[^[:digit:]])` ensures save navigation does not match
2. match the ternary question mark with `(\?)`
3. negative lookahead after the first question mark to ensure we don't match nullish coalesce `(?!\?)`

If you don't want to add those here, could you add comments like this to the ternary test cases?

Reply:
Done